### PR TITLE
docs: Add example how to config ipmasq via ConfigMap

### PR DESCRIPTION
**Pull request by brb**

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
Looks good in principle, should we include this under `examples/kubernetes`?

**Comment by joestringer** [comment]
Yeah, so then you can do one-liners like this for a common configuration (eg RFC1918 address ranges):

```
kubectl apply -f https://raw.githubusercontent.com/cilium/cilium/master/examples/minikube/sw_l3_l4_policy.yaml
```

With this, we can do `.. include: /path/to/ipmasq-config.yaml` in the docs to still render it the same as is done currently in this PR.

Additionally, for testing it means that the file is in the tree so it's easy to apply and then observe some changes.

**Review by joestringer — APPROVED** [review]
Thanks!

File: Documentation/concepts/networking/masquerading.rst [hunk 1]
```diff
@@ -116,23 +116,17 @@ The agent uses Fsnotify to track updates to the configuration file, so the origi
...
-    - 172.16.0.0/12
-    - 192.168.0.0/16
-    masqLinkLocal: false
+.. parsed-literal::
 
-    $ kubectl create configmap ip-masq-agent --from-file=agent-config --namespace=kube-system
+    $ kubectl create -n kube-system -f \ |SCM_WEB|\/examples/kubernetes-ip-masq-agent/rfc1918.yaml
 
-    $ # Wait ~60s until the ConfigMap is mounted into a cilium pod
+    # Wait ~60s until the ConfigMap is propagated into the configuration file
```
Comment (joestringer) [inline Documentation/concepts/networking/masquerading.rst]: Actually this was intentional before: `$` at start of line is user shell prompt, `#` is root shell prompt; this is intended to be `$ #` which is a comment inside a user shell prompt.
```suggestion
    $ # Wait ~60s until the ConfigMap is propagated into the configuration file
```

Comment (joestringer) [reply Documentation/concepts/networking/masquerading.rst]: ^^ Probably doesn't matter so much though, this is a shell+output example so you can't just copy the whole thing and run it anyway. :ship:

**Comment by joestringer** [comment]
The BPF check does complain about the misformatted signoff though, please fix that.

**Comment by brb** [comment]
> The BPF check does complain about the misformatted signoff though, please fix that.

Done.

File: Documentation/concepts/networking/masquerading.rst (hunk 1 above)
Comment (brb) [reply Documentation/concepts/networking/masquerading.rst]: Ah, let me bring it back then to be consistent.

**Comment by aanm** [comment]
@brb I assume this PR is meant to be backported to 1.12. Please remove the label if that is not the case. Thank you

**Pull request merged**